addons/common/functions/fn_serializeWaypoint.sqf

```diff
@@ -1,16 +1,30 @@
-params ["_group","_waypointIndex"];
-private _way = [];
-_way pushBack waypointName [_group,_waypointIndex];
-_way pushBack waypointPosition  [_group,_waypointIndex];
-_way pushBack waypointType [_group,_waypointIndex];
-_way pushBack waypointBehaviour [_group,_waypointIndex];
-_way pushBack waypointCombatMode [_group,_waypointIndex];
-_way pushBack waypointDescription [_group,_waypointIndex];
-_way pushBack waypointFormation [_group,_waypointIndex];
-_way pushBack waypointHousePosition [_group,_waypointIndex];
-_way pushBack waypointScript [_group,_waypointIndex];
-_way pushBack waypointShow [_group,_waypointIndex];
-_way pushBack waypointSpeed [_group,_waypointIndex];
-_way pushBack waypointTimeout [_group,_waypointIndex];
-_way pushBack waypointVisible [_group,_waypointIndex];
-_way
+/*
+ * Name: TMF_common_fnc_serializeWaypoint
+ * Author: Head
+ *
+ * Arguments:
+ * _this: Waypoint
+ *
+ * Return:
+ * array: waypoint data
+ *
+ */
+private _way = [
+    waypointName _this,
+    waypointPosition  _this,
+    waypointType _this,
+    waypointBehaviour _this,
+    waypointCombatMode _this,
+    waypointDescription _this,
+    waypointFormation _this,
+    waypointHousePosition _this,
+    waypointScript _this,
+    waypointSpeed _this,
+    waypointTimeout _this,
+    waypointVisible _this,
+    waypointCompletionRadius _this,
+    waypointStatements _this,
+    waypointAttachedObject _this,
+    waypointAttachedVehicle _this
```

Comment:
The only option that would be any semblance of more performant would be storing nil I guess?
Since its an index based data structure all the items need to be there.

I am open to suggestions.

Reply:
You could have a look at what I did here https://github.com/TMF3/TMF/blob/5ef16e3fa29485912f074e1c6f89e55b0751e46b/addons/assigngear/functions/fn_assignGear.sqf#L41-L147